File: argentum-api/api/tests/data/tag_registrations.py
from django.utils.dateparse import parse_datetime

from api.models import TagRegistration
from api.tests.data.guests import TestGuests
from api.tests.data.orders import TestOrders
from api.tests.utils.test_objects import TestObjects


class TestTagRegistrations(TestObjects):
    MODEL = TagRegistration

    ROBY_TWO: MODEL
    SHEELAH_THREE: MODEL
    ROBY_FOUR: MODEL

    ROBY_FOUR_COMMITTED: MODEL
    ROBY_FIVE: MODEL
    ROBY_THREE_STEAL: MODEL
    ROBY_THREE_STEAL_COMMITTED: MODEL

    @classmethod
    def init(cls):
        cls.ROBY_TWO = TagRegistration(
            id=24010,
            time=parse_datetime('2019-12-31T22:07:01Z'),
            guest=TestGuests.ROBY,
            order=TestOrders.TAG_REGISTRATION_TWO,
            pending=False
        )

        cls.SHEELAH_THREE = TagRegistration(
            id=24020,
            time=parse_datetime('2019-12-31T22:09:01Z'),
            guest=TestGuests.SHEELAH,
            order=TestOrders.TAG_REGISTRATION_THREE,
            pending=False
        )

        cls.ROBY_FOUR = TagRegistration(
            id=24030,
            time=parse_datetime('2019-12-31T22:15:01Z'),
            guest=TestGuests.ROBY,
            order=TestOrders.TAG_REGISTRATION_FOUR,
            pending=True
        )

        cls.SAVED = [cls.ROBY_TWO, cls.SHEELAH_THREE, cls.ROBY_FOUR]

        cls.ROBY_FOUR_COMMITTED = TagRegistration(
            id=24031,
            guest=TestGuests.ROBY,
            order=TestOrders.TAG_REGISTRATION_FOUR,
            pending=False
        )

        cls.ROBY_FIVE = TagRegistration(
            id=24040,
            guest=TestGuests.ROBY,
            order=TestOrders.TAG_REGISTRATION_FIVE,
            pending=True
        )

        cls.ROBY_THREE_STEAL = TagRegistration(
            id=24050,
            guest=TestGuests.ROBY,
            order=TestOrders.TAG_REGISTRATION_FOUR,
            pending=True
        )

        cls.ROBY_THREE_STEAL_COMMITTED = TagRegistration(
            id=24051,
            guest=TestGuests.ROBY,
            order=TestOrders.TAG_REGISTRATION_FOUR,
            pending=False
        )

        cls.UNSAVED = [cls.ROBY_FOUR_COMMITTED, cls.ROBY_FIVE, cls.ROBY_THREE_STEAL, cls.ROBY_THREE_STEAL_COMMITTED]

File: argentum-ui/src/app/common/rest-service/test-data/tag-registrations.ts
import {TagRegistration} from '../../model/tag-registration';
import {Tags} from './tags';
import {Orders} from './orders';

export namespace TagRegistrations {
  export const ROBY_TWO = new TagRegistration(
    24010,
    new Date('2019-12-31T22:07:01Z'),
    [Tags.TWO.label],
    undefined,
    undefined,
    Orders.TAG_REGISTRATION_TWO,
    false
  );

  export const SHEELAH_THREE = new TagRegistration(
    24020,
    new Date('2019-12-31T22:09:01Z'),
    [Tags.THREE.label],
    undefined,
    undefined,
    Orders.TAG_REGISTRATION_THREE,
    false
  );

  export const ROBY_FOUR = new TagRegistration(
    24030,
    new Date('2019-12-31T22:15:01Z'),
    [Tags.FOUR.label],
    undefined,
    undefined,
    Orders.TAG_REGISTRATION_FOUR,
    true
  );

  export const ALL = [ROBY_TWO, SHEELAH_THREE];

  export const ROBY_FOUR_COMMITTED_REFERENCE = new TagRegistration(
    24031,
    new Date('2019-12-31T22:15:01Z'),
    [Tags.FOUR.label],
    undefined,
    undefined,
    Orders.TAG_REGISTRATION_FOUR,
    false
  );

  export const ROBY_FIVE_REFERENCE = new TagRegistration(
    24040,
    new Date('2019-12-31T22:19:01Z'),
    [Tags.FIVE.label],
    undefined,
    undefined,
    Orders.TAG_REGISTRATION_FIVE,
    true
  );
}
